### PR TITLE
improve login flow

### PR DESCRIPTION
**Pull request by johannesauer**
- create custom view for lock/unlock animation
- unify setup/login unlock presentation
- add lock menu item enable/disable logic

### CODE REVIEW

File: macos/user interface/LoginUserInterface.swift [hunk 1]
```diff
@@ -1,12 +1,11 @@
 
 import AppKit
 
-class LoginUserInterface {
+class LoginUserInterface: NSObject {
     
-    let root = BackgroundView()
+    let lockerView = LockerView()
     let lockIconView = NSImageView()
```
Comment (NicoSa) [inline macos/user interface/LoginUserInterface.swift]: shutterView is probably better

File: macos/user interface/SetupUserInterface.swift [hunk 2]
```diff
@@ -17,37 +16,29 @@ extension SetupUserInterface: UserInterface {
     typealias Responder = SetupController
     
     func setupViewHierarchy(responder: Responder) {
-        root.addConstrained(subview: lockIconView)
-        root.addConstrained(subview: passwordField)
-        root.addConstrained(subview: repeatPasswordField)
-        root.addConstrained(subview: createVaultButton)
-        root.addLayoutGuide(container)
+        lockerView.topCurtain.addConstrained(subview: lockIconView)
+        lockerView.bottomCurtain.addConstrained(subview: passwordField)
```
Comment (NicoSa) [inline macos/user interface/SetupUserInterface.swift]: bottomShutter and topShutter

File: macos/view/LockerView.swift [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    
+    private let overlayView = BackgroundView()
+    private let overlayAlphaValue = CGFloat(0.2)
+    private let animationDuration = TimeInterval(0.5)
+    
+    let topCurtain = BackgroundView()
+    let bottomCurtain = BackgroundView()
+    
+    var topCurtainTopContraint: NSLayoutConstraint?
+    var topCurtainCenterContraint: NSLayoutConstraint?
```
Comment (NicoSa) [inline macos/view/LockerView.swift]: replace Curtain with Shutter

File: macos/view/LockerView.swift [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        fatalError()
+    }
+    
+    func unlock(completion: @escaping () -> Void) {
+        overlayView.alphaValue = overlayAlphaValue
+        topCurtainCenterContraint?.isActive = true
+        topCurtainTopContraint?.isActive = false
+        bottomCurtainCenterConstraint?.isActive = true
+        bottomCurtainBottomConstraint?.isActive = false
+
```
Comment (NicoSa) [inline macos/view/LockerView.swift]: extra lines

File: macos/user interface/LoginUserInterface.swift (hunk 1 above)
Comment (johannesauer) [reply macos/user interface/LoginUserInterface.swift]: done

File: macos/user interface/SetupUserInterface.swift (hunk 2 above)
Comment (johannesauer) [reply macos/user interface/SetupUserInterface.swift]: done

File: macos/view/LockerView.swift (hunk 3 above)
Comment (johannesauer) [reply macos/view/LockerView.swift]: done

File: macos/view/LockerView.swift (hunk 4 above)
Comment (johannesauer) [reply macos/view/LockerView.swift]: done

**Pull request merged**